### DISCUSSION TITLE
Simple chat interface

### DISCUSSION THREAD

**Discussion by cachapa** [onepub-dev/dcli#220] [Q&A] [answered]
I'm building a sample CLI chat app for a new package, and I need a very basic UI that simply allows the user to type their message at the bottom of the screen, while seeing messages appear above it.

Essentially it should look something like this:

```
[John] Hello all
[Jane] Good morning
> How's everyon
```

The world of CLI formatting tools is a bit overwhelming, so I was hoping that someone can guide me towards a simple solution. I don't really need to take over the entire terminal, or to nail down usability since it's just a demo app.

**Comment by bsutton** [comment]
Have a look at this doco.
https://dcli.onepub.dev/dcli-api/displaying-information

**Comment by cachapa** [comment]
Thanks. From reading those docs it seems like I should be using the `row` operator to position the cursor whenever a new message arrives.

I'll have to experiment and see if I can get it to work without making it so complex it detracts from the actual code I'm trying to demonstrate :-)

**Comment by bsutton** [comment, marked as answer]
You probably need to send a newline to the console to make things scroll up
and then reprint the prompt.
